ui/src/main/java/com/cloudinary/android/uploadwidget/ui/UploadWidgetActivity.java

```diff
@@ -72,7 +72,7 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
                 ArrayList<Uri> uris = extractImageUris(data);
 
                 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-                    final int takeFlags = data.getFlags() & (Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
+                    final int takeFlags = data.getFlags() & (Intent.FLAG_GRANT_READ_URI_PERMISSION);
```

Comment:
I'm guessing the original flag name wasn't an accident - does this drop support for older versions of android?

Reply:
It's related to the intent type
Since on the bottom file `UploadWidget.java` I changed the intent type I hard to change the permission file as well, they go together hand in hand.
The old way, using ACTION_GET_COTNENT saves the uri for a short time, and we get a permission denial error.